datafusion/sql/src/select.rs

```diff
@@ -766,9 +769,9 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
             check_columns_satisfy_exprs(
```

Comment:
We can have `check_columns_satisfy_exprs_for_having_clause` if the logic is different for projection and having

Reply:
I think the function can be shared. The difference is that the check of `having` requires a return value to confirm that the validation has been executed and is valid.